313 Multiple form types for a single widget

juzen2003 opened the pull request:
- The is the pull request to track issue: #313 
- Implementation done:
    - phase 1: Allow sub-categories in checkbox fields. Allow multiple levels of sub-category.
        - Now ```grouping``` field can store directory tree like main_cat/sub_cat1/sub_cat2  

- Steps to test (phase 1):
1. In config_targets.py, replace the normal PLANET_GROUP_MAPPING to the commented out one and re-import the test database.
2. Intended target name widget will have multiple level of sub categories shown. Each level can have a mix of checkboxes and sub-category.

juzen2003 committed:
- Update _update_form_with_grouping to handle multiple categories/sub-categories
  of forms in the search widget.
- Add (commented out) testing PLANET_GROUP_MAPPING for testing multiple
  levels of categories in intended target name widget.
- Update css to align checkboxes and "+" icons for expanding/collapsing
  categories.

rfrenchseti reviewed (changes requested):
The green hinting numbers are wrong. I'm not quite sure how they're being calculated now but they don't make sense. The numbers at each level should include ALL of the checkboxes at all levels below it.

juzen2003 committed:
Fixed the green hints for category/sub-category/checkboxes by using hierarchy
tree. Previously we are using the directory name, and different categories
can have the same directory name which will sum up all the counts even
if the hierarchy tree is different.
